lib/cheetah.rb

```diff
@@ -375,6 +378,7 @@ class << self
 
     def run(*args)
       options = args.last.is_a?(Hash) ? args.pop : {}
+      return changed_env(options, *args) if options[:env]
```

Comment:
From the top of my head (sure there are better approaches if we put some thinking into it)
```ruby 
def run(*args)
  options = args.last.is_a?(Hash) ? args.pop : {}
  env = options.delete(:env)
  with_env(env) do
    #rest of the run code
  end
end

private

def with_env(env)
  old_env = ENV.to_hash
  ENV.update(env)
  yield
ensure
  ENV.replace(old_env)
end
```

Sure it will not work as it is and will need some work, but you get the idea :wink: